x-pack/platform/packages/shared/kbn-streamlang-tests/test/scout/api/tests/cross_compatibility/append.spec.ts

```diff
@@ -0,0 +1,170 @@
+            expect(() => transpileEsql(streamlangDSL)).toThrow(); // ES|QL transpiler should reject
+          }
+        );
+      });
+    });
+
+    streamlangApiTest.describe('Incompatible', () => {
+      streamlangApiTest(
+        'should result in an array in ingest, but a scalar in ES|QL for a single value',
+        async ({ testBed, esql }) => {
+          const streamlangDSL: StreamlangDSL = {
+            steps: [
+              {
+                action: 'append',
+                to: 'tags',
+                value: ['new_tag'],
+              } as AppendProcessor,
+            ],
+          };
+
+          const { processors } = transpileIngestPipeline(streamlangDSL);
+          const { query } = transpileEsql(streamlangDSL);
+
+          const docs = [{ message: 'a' }];
+          await testBed.ingest('ingest-append-scalar', docs, processors);
+          const ingestResult = await testBed.getDocsOrdered('ingest-append-scalar');
+
+          // ES|QL requires the field to be pre-mapped, so we ingest a doc with the field first
+          await testBed.ingest('esql-append-scalar', [{ tags: ['initial_tag'] }, ...docs]);
+          const esqlResult = await esql.queryOnIndex('esql-append-scalar', query);
+
+          expect(ingestResult[0].tags).toEqual(['new_tag']); // Ingest creates an array
+          expect(esqlResult.documentsOrdered[1].tags).toEqual('new_tag'); // ES|QL creates a scalar
+        }
+      );
+
+      streamlangApiTest(
+        'should result in an array in ingest, but a scalar in ES|QL when deduplicating to a single value',
+        async ({ testBed, esql }) => {
+          const streamlangDSL: StreamlangDSL = {
+            steps: [
+              {
+                action: 'append',
+                to: 'tags',
+                value: ['existing_tag'],
+                allow_duplicates: false,
+              } as AppendProcessor,
+            ],
+          };
+
+          const { processors } = transpileIngestPipeline(streamlangDSL);
+          const { query } = transpileEsql(streamlangDSL);
+
+          const docs = [{ tags: ['existing_tag'] }];
+          await testBed.ingest('ingest-append-dedupe', docs, processors);
+          const ingestResult = await testBed.getDocsOrdered('ingest-append-dedupe');
+
+          await testBed.ingest('esql-append-dedupe', docs);
+          const esqlResult = await esql.queryOnIndex('esql-append-dedupe', query);
+
+          expect(ingestResult[0].tags).toEqual(['existing_tag']); // Ingest results in an array
+          expect(esqlResult.documentsOrdered[0].tags).toEqual('existing_tag'); // ES|QL results in a scalar
+        }
+      );
+    });
+
+    // Note that the Incompatible test suite doesn't necessarily mean the features are functionally incompatible,
+    // rather it highlights the nuanced behavioral differences in certain edge cases among transpilers.
+    streamlangApiTest.describe('Incompatible', () => {});
```

Comment:
`question` : what is the purpose if it? I'm afraid this empty suite might be misinterpreted in the events report due to 0 run time.